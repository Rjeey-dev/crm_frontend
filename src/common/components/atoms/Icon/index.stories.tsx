import React from 'react';
import { storiesOf } from '@storybook/react';
import Icon from './index';

storiesOf('Icon', module)
    .add('notifications', () => (
        <Icon classes='mdi mdi-file-document-box'/>
    ))
    .add('settings', () => (
        <Icon classes='mdi mdi-tune'/>
    ));