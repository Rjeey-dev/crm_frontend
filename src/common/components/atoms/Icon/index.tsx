import React from 'react';

interface IProps {
    classes: string,
    attributes?: object
}

function Icon(props: IProps) {
    return <i className={props.classes} {...props.attributes}/>;
}

export default Icon;
