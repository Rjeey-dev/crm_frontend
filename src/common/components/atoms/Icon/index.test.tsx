import { shallow } from 'enzyme';
import React from 'react';

import Icon from './index';

it("Renders <Icon/> with correct properties", () => {
    const output = shallow(
        <Icon classes='class'/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Icon/> with attributes", () => {
    const output = shallow(
        <Icon classes='class' attributes='test'/>
    ).html();

    expect(output).toMatchSnapshot();
});
