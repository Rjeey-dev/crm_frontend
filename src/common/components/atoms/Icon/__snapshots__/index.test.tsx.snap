// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Icon/> with attributes 1`] = `"<i class=\\"class\\"></i>"`;

exports[`Renders <Icon/> with correct properties 1`] = `"<i class=\\"class\\"></i>"`;
